examples/cpp/dds/BasicConfigurationExample/BasicConfiguration_main.cpp

```diff
@@ -60,154 +60,155 @@ int main(
-                    return 1;
-                    break;
-            }
-        }
-        if (transient && !reliable)
-        {
-            std::cerr << "WARNING: --transient will take no effect since not reliable." << std::endl;
-        }
     }
     else
     {
-        std::cerr << "ERROR: <publisher|subscriber> argument is required." << std::endl;
+        std::cerr << "ERROR: first argument can only be <publisher|subscriber>" << std::endl;
         option::printUsage(fwrite, stdout, usage, columns);
         return 1;
     }
 
+    for (int i = 0; i < parse.optionsCount(); ++i)
+    {
+        option::Option& opt = buffer[i];
+        switch (opt.index())
+        {
+            case optionIndex::HELP:
+                // not possible, because handled further above and exits the program
+                break;
+
+            case optionIndex::TOPIC:
+                topic_name = std::string(opt.arg);
+                break;
+
+            case optionIndex::DOMAIN_ID:
+                domain = strtol(opt.arg, nullptr, 10);
+                break;
+
+            case optionIndex::SAMPLES:
+                count = strtol(opt.arg, nullptr, 10);
+                break;
+
+            case optionIndex::INTERVAL:
+                if (type == PUBLISHER)
+                {
+                    sleep = strtol(opt.arg, nullptr, 10);
+                }
+                else
+                {
+                    print_warning("publisher", opt.name);
+                }
+                break;
+
+            case optionIndex::WAIT:
+                if (type == PUBLISHER)
+                {
+                    num_wait_matched = strtol(opt.arg, nullptr, 10);
+                }
+                else
+                {
+                    print_warning("publisher", opt.name);
+                }
+                break;
+
+            case optionIndex::ASYNC:
+                if (type == PUBLISHER)
+                {
+                    async = true;
+                }
+                else
+                {
+                    print_warning("publisher", opt.name);
+                }
+                break;
+
+            case optionIndex::TRANSPORT:
+                if (strcmp(opt.arg, "shm") == 0)
+                {
+                    transport = SHM;
+                }
+                else if (strcmp(opt.arg, "udp") == 0 || (strcmp(opt.arg, "udpv4") == 0))
+                {
+                    transport = UDPv4;
+                }
+                else if (strcmp(opt.arg, "udpv6") == 0)
+                {
+                    transport = UDPv6;
+                }
+                break;
+
+            case optionIndex::RELIABLE:
+                reliable = true;
+                break;
+
+            case optionIndex::TRANSIENT_LOCAL:
+                transient = true;
+                break;
+
+            case optionIndex::TTL:
+                hops = strtol(opt.arg, nullptr, 10);
```

Comment:
Maybe add a message warning that this argument will take no effect when transport is `SHM`.

Also, it would be good to have a warning somewhere, or in the help menu, informing about the `255` hard limit.

Reply:
fixed